test/SlowTests/Server/RecordingTransactionOperationsMergerTests.cs

```diff
@@ -0,0 +1,509 @@
+            };
+
+            using (var store = GetDocumentStore())
+            {
+                store.Maintenance.Send(new StartTransactionsRecordingOperation(filePath));
+
+                using (var session = store.OpenSession())
+                using (var picture = File.Open(@"C:\Records\Igal.jpg", FileMode.Open))
+                {
+                    session.Store(album, "albums/1");
+
+                    session.Advanced.Attachments.Store("albums/1", "Igal.jpg", picture, "image/jpeg");
+
+                    session.SaveChanges();
+
+                }
+                store.Maintenance.Send(new StopTransactionsRecordingOperation());
+            }
+
+            using (var store = GetDocumentStore())
+            using (var replayStream = new FileStream(filePath, FileMode.Open))
+            {
+                store.Maintenance.Send(new ReplayTransactionsRecordingOperation(replayStream));
+            }
+        }
+
+        [Fact]
+        public void StartRecordingWithoutStop_ShouldResultInLegalJson()
+        {
+            var filePath = Path.GetTempFileName();
+
+            //Recording
+            using (var store = GetDocumentStore())
+            {
+                store.Maintenance.Send(new StartTransactionsRecordingOperation(filePath));
+            }
+
+            //Todo to change to zip after recording will change to zip
+            //            using (var fileStream = File.OpenRead(filePath))
+            //            using (var zippedStream = new GZipStream(fileStream, CompressionMode.Decompress))
+            //            using (var reader = new StreamReader(zippedStream))
+            //            {
+            //                var fileContent = File.ReadAllText(filePath);
+            //                JToken.Parse(fileContent);
+            //            }
+            var fileContent = File.ReadAllText(filePath);
+            JToken.Parse(fileContent);
+        }
+
+        [Fact]
+        public void RecordingDeleteCommand()
+        {
+            var filePath = Path.GetTempFileName();
+
+            var expectedNames = new[] { "Avi" };//, "Baruch", "Charlotte", "Dylan", "Eli", "Fabia", "George", "Harper" };
+            var users = expectedNames.Select(n => new User { Name = n }).ToArray();
+            User[] documents = { };
+
+
+            //Recording
+            using (var store = GetDocumentStore())
+            {
+                store.Maintenance.Send(new StartTransactionsRecordingOperation(filePath));
+                using (var session = store.OpenSession())
+                {
+                    foreach (var user in users)
+                    {
+                        session.Store(user);
+                    }
+                    session.SaveChanges();
+
+                    store.Commands().Execute(new DeleteDocumentCommand(users.First().Id, null));
+                }
+                store.Maintenance.Send(new StopTransactionsRecordingOperation());
```

Comment:
Probably need to wait until this is actually being completed.

Reply:
Do you mean I need to add a possibility to wait to "StopTransactionsRecordingOperation"?
And if yes should I do this to "StartTransactionsRecordingOperation" & "ReplayTransactionsRecordingOperation" as well?

Reply:
In progress